x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -281,7 +290,10 @@ private static Version readMappingVersion(String indexName, MappingMetaData mapp
      */
     public void checkIndexVersionThenExecute(final Consumer<Exception> consumer, final Runnable andThen) {
         final State indexState = this.indexState; // use a local copy so all checks execute against the same state!
-        if (indexState.indexExists && indexState.isIndexUpToDate == false) {
+        if (indexState.concreteIndexName == null) {
```

Comment:
use `stateRecovered() == false` ?

Reply:
I went with your suggestion, and then reverted. I have kept on checking `indexState` because this is the "frozen" index state that is used in all tests.